src/lib/core/CHIPConfig.h

```diff
@@ -509,12 +509,22 @@
 /**
  *  @def CHIP_CONFIG_ERROR_SOURCE
  *
- *  If asserted (1), then CHIP_ERROR constants will include the source location of their expansion.
+ *  If asserted (1), then CHIP_ERROR constants will include the minimum source location (file, line) of their expansion.
  */
 #ifndef CHIP_CONFIG_ERROR_SOURCE
 #define CHIP_CONFIG_ERROR_SOURCE 0
 #endif // CHIP_CONFIG_ERROR_SOURCE
 
+/**
+ *  @def CHIP_CONFIG_ERROR_SOURCE_FULL
+ *
+ *  If asserted (1), then CHIP_ERROR constants will include the std::source_location of their expansion.
+ *  Note: this option significantly increases output binary size
+ */
+#ifndef CHIP_CONFIG_ERROR_SOURCE_FULL
+#define CHIP_CONFIG_ERROR_SOURCE_FULL 0
```

Comment:
The current SDK behavior is `1` by default so this changes the behavior in a worst way. It needs to be enabled by default for POSIX platform. This must be tied to BUILD.gn command line arguments.

Reply:
The fact that it was enabled in https://github.com/project-chip/connectedhomeip/pull/32935 - was a regression. 300Kb - is a very high cost for some additional info that even not used inside CHIP.  I don't think it should be enabled by default

Reply:
how large are posix platforms by default? I believe those we generally consider "large devices" so we generally do not pay as much attention to their side as for embedded devices.

Being able to see better errors/stacktraces where the binary size is acceptable seems preferred. Likely not used within CHIP because CHIP needs to compile with C++17.

Reply:
Let's make it configurable for now, but do not change the default (i.e. keep it enabled to not change behavior). Then we can argue about the default in a separate PR. 

I believe `C++20` is indeed not ideal as a detection mechanism: we generally want it on large systems (say linux, maybe darwin, android) but do not want it on small systems. How to make that decision needs to be determined. BUILD.gn config sounds the best.

Reply:
> how large are posix platforms by default? I believe those we generally consider "large devices" so we generally do not pay as much attention to their side as for embedded devices.

In our case final binary size is about 20Mb, so 300Kb is a noticeable difference. Well, it is about 375, but those 75 are understandable as we are migrating from 1.2 to 1.4. But these 300Kb - is a payment for nothing as file/line seems enough for debugging.

But ok. If you say thats not a regression, I will switch default value to 1

Reply:
Take a look now. I've enabled it on platforms where `CHIP_CONFIG_ERROR_SOURCE` is enabled

Reply:
> Please place it in a build_config_header rule so that we can get platforms to override this. This was added for it to be used explicitly, so we want to provide the option of "keep it on" for people really needing this.

What is build_config_header rule?

Reply:
Do you mean https://github.com/project-chip/connectedhomeip/blob/master/src/platform/BUILD.gn#L113 ?

Thats a little bit difficult for `1` as a default value. As it depends on compiler version. I've placed default value for `CHIP_CONFIG_ERROR_SOURCE_FULL` near the `CHIP_CONFIG_ERROR_SOURCE`

Reply:
Let's put this configuration as a `declare_args` in our builds and have it part of a https://github.com/project-chip/connectedhomeip/blob/master/build/chip/buildconfig_header.gni#L107 setup.

Reply:
Core has one such header: https://github.com/project-chip/connectedhomeip/blob/master/src/lib/core/BUILD.gn#L25

Reply:
For compiler version I am unsure ... maybe it should be "allow_...." and then the real check validates both allow and if the compiler supports it (although realistically maybe we should default to 0 and reconsider later when we use a newer compiler ...)

Reply:
In fact 0 default looks more straightforward: enable if you know what are you doing and be sure that your compiler supports it.
At least until default C++ version is 17

Reply:
Please place it in the core config and we can default it to 0 for now in declare_args. 
If others need it it can be enabled. however it should be a integrator choice rather than a platform  default (i.e. you could compile for linux and still choose to have it or not)